src/ProjectTemplates/Shared/Project.cs

```diff
@@ -151,23 +151,41 @@ internal async Task<ProcessResult> RunDotNetPublishAsync(IDictionary<string, str
 
         var restoreArgs = noRestore ? "--no-restore" : null;
 
-        using var result = ProcessEx.Run(Output, TemplateOutputDir, DotNetMuxer.MuxerPathOrDefault(), $"publish {restoreArgs} -c Release /bl {additionalArgs}", packageOptions);
-        await result.Exited;
-        CaptureBinLogOnFailure(result);
-        return new ProcessResult(result);
+        using var execution = ProcessEx.Run(Output, TemplateOutputDir, DotNetMuxer.MuxerPathOrDefault(), $"publish {restoreArgs} -c Release /bl {additionalArgs}", packageOptions);
+        await execution.Exited;
+
+        var result = new ProcessResult(execution);
+
+        // Fail if there were build warnings
```

Comment:
Do you know when we ignoring errorOnBuildWarning today?

Reply:
Hmm it's possible I'm being too presumptive. The original issue that prompted this PR was due to a compiler warning on the new project so I assumed we weren't failing tests properly if there were warnings. It's possible that instead the case was just completely missing and this change isn't needed.

Reply:
No I like that its failing if there are errors/warnings, I guess I'm asking if we need this to be configurable, if noone is setting it to false, can we just always do this?  Silently ignoring build warnings seems... bad?

Reply:
Yeah OK, I can just remove the optionality then.